data/transactions/logic/resources_test.go

```diff
@@ -153,6 +164,18 @@ func TestAppSharing(t *testing.T) {
 	sources = []string{"", "", "gtxn 1 Accounts 1; gtxn 0 Applications 0; byte 0xAA; app_local_get_ex"}
 	logic.TestApps(t, sources, txntest.Group(&appl0, &appl1, &appl2), 9, nil,
 		logic.Exp(2, "unavailable Local State")) // note that the error message is for Locals, not specialized
+
+	// try to do a put on local state of the account in tx1, but tx0 ought not have access to that local state
+	ledger.NewAccount(pay1.Sender, 200_000)
+	ledger.NewLocals(pay1.Sender, 900) // opt in
+	sources = []string{`gtxn 1 Sender; byte "key"; byte "val"; app_local_put; int 1`}
+	logic.TestApps(t, sources, txntest.Group(&appl0, &pay1), 9, ledger,
+		logic.Exp(0, "unavailable Local State "+pay1.Sender.String()))
+
+	// same, do app_local_del
+	sources = []string{`gtxn 1 Sender; byte "key"; app_local_del; int 1`}
+	logic.TestApps(t, sources, txntest.Group(&appl0, &pay1), 9, ledger,
+		logic.Exp(0, "unavailable Local State "+pay1.Sender.String()))
```

Comment:
if I read correctly, both cases should fail for not putting `pay1.Sender` in one of the foreign array in txn-group. Now that the address is not in `cx.available.sharedLocals`, thus `allowsLocals` error?

(which is testing the same thing as above, but pointing out `app_local_put/del`?

Reply:
yes, same test but for `_del`.

I'm fixing the grammar in the comment.